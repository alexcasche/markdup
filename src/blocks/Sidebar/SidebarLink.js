import { applyStyleModifiers } from 'styled-components-modifiers'
import { tint } from 'polished'
import { Link } from 'elements'

const MODIFIERS = {
  active: ({ theme }) => `
    &, &:hover {
      pointer-events: none;
      cursor: default;
      color: ${theme.colors.grey};
      svg.fa-check-circle {
        color: ${theme.colors.grey};
      }
    }
  `,
}

const SidebarLink = Link.extend`
  position: relative;
  display: block;
  padding: 0 0 0 50px;
  border-bottom: 1px solid ${props => tint(0.85, props.theme.colors.black)};
  color: ${props => props.theme.colors.grey};
  font-size: 14px;
  font-weight: 400;
  line-height: 50px;
  text-align: left;
  text-decoration: none;
  letter-spacing: 1px;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  transition: color 0.15s ease-in-out;
  &:last-child {
    border-bottom: none;
  }
  svg {
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    font-size: 15px;
    transition: inherit;
    &.fa-check-circle {
      color: transparent;
    }
    &.fa-link {
      font-size: 14px;
    }
  }
  &:hover {
    cursor: pointer;
    color: ${props => props.theme.colors.green};
    svg {
      color: ${props => props.theme.colors.green};
    }
    .fa-check-circle {
      color: transparent;
    }
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default SidebarLink
