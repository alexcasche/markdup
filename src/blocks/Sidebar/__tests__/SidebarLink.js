import SidebarLink from '../SidebarLink'
import { shade } from 'polished'

describe('SidebarLink Element', () => {
  it('renders correctly', () => {
    const tree = render(<SidebarLink />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds active modifier', () => {
    const tree = render(<SidebarLink modifiers={['active']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('pointer-events', 'none')
  })
})
