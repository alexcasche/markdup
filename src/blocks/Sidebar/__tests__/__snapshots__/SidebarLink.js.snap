// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLink Element adds active modifier 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  position: relative;
  display: block;
  padding: 0 0 0 50px;
  border-bottom: 1px solid #3e4042;
  color: #ABADB3;
  font-size: 14px;
  font-weight: 400;
  line-height: 50px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c0:hover {
  cursor: pointer;
}

.c0:last-child {
  border-bottom: none;
}

.c0 svg {
  position: absolute;
  right: 20px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  font-size: 15px;
  -webkit-transition: inherit;
  transition: inherit;
}

.c0 svg.fa-check-circle {
  color: transparent;
}

.c0 svg.fa-link {
  font-size: 14px;
}

.c0:hover {
  cursor: pointer;
  color: #34F96E;
}

.c0:hover svg {
  color: #34F96E;
}

.c0:hover .fa-check-circle {
  color: transparent;
}

.c0,
.c0:hover {
  pointer-events: none;
  cursor: default;
  color: #ABADB3;
}

.c0 svg.fa-check-circle,
.c0:hover svg.fa-check-circle {
  color: #ABADB3;
}

<a
  className="c0"
/>
`;

exports[`SidebarLink Element renders correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  position: relative;
  display: block;
  padding: 0 0 0 50px;
  border-bottom: 1px solid #3e4042;
  color: #ABADB3;
  font-size: 14px;
  font-weight: 400;
  line-height: 50px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c0:hover {
  cursor: pointer;
}

.c0:last-child {
  border-bottom: none;
}

.c0 svg {
  position: absolute;
  right: 20px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  font-size: 15px;
  -webkit-transition: inherit;
  transition: inherit;
}

.c0 svg.fa-check-circle {
  color: transparent;
}

.c0 svg.fa-link {
  font-size: 14px;
}

.c0:hover {
  cursor: pointer;
  color: #34F96E;
}

.c0:hover svg {
  color: #34F96E;
}

.c0:hover .fa-check-circle {
  color: transparent;
}

<a
  className="c0"
/>
`;
